loclist.kak

```diff
@@ -0,0 +1,147 @@
+## Location List
+
+declare-option -hidden str loclist_buffer
+declare-option -hidden str-list loclist_stack
+declare-option -docstring %{Defines location patterns for location lists.
+Default locations look like "file:line[:column][:message]"
+
+Capture groups must be:
+    1: filename
+    2: line number
+    3: optional column
+    4: optional message
+} regex loclist_location_format ^\h*\K([^:\n]+):(\d+)\b(?::(\d+)\b)?(?::([^\n]+))
+
+declare-option str loclist_root ''
+
+define-command loclist-jump %{ # from grep.kak
+    evaluate-commands -save-regs abc %{ # use evaluate-commands to ensure jumps are collapsed
+        try %{
+            evaluate-commands -draft -save-regs / %{
+                set-register / %opt{loclist_location_format}
+                execute-keys <semicolon>xs<ret>
+                set-register a "%reg{1}"
+                set-register b "%reg{2}"
+                set-register c "%reg{3}"
+                try %{
+                    # Is it an absolute path?
+                    execute-keys s\A/.*<ret>
+                } catch %{
+                    set-register a "%opt{loclist_root}%reg{a}"
+                }
+            }
+            set-option buffer grep_current_line %val{cursor_line}
```

Comment:
We could move away from `grep_current_line` to `locations_current_line`.

Our variable would then be used by all of make/grep/lint/lsp once we merge it upstream.
Until then, we can keep both variables around and synchronize them with WinSetOption hooks. If that's too complex let's not worry about it for now.

Of course there is an argument for keeping grep_current_line:
`:grep` is probably the consumer in ~80% of all cases (except if supplanted by `:lsp-references`).

If we stop depending on grep, then we should also declare `jumpclient` here.

Reply:
If this is upstreamed, i think grep.kak should be reduced to essentially just the one command. So yes, all of this should be locations_current_line etc